storage/innobase/lock/lock0lock.cc

```diff
@@ -4591,22 +4660,70 @@ static bool lock_release_on_prepare_try(trx_t *trx)
       bool supremum_bit= lock_rec_get_nth_bit(lock, PAGE_HEAP_NO_SUPREMUM);
       bool rec_granted_exclusive_not_gap=
         lock->is_rec_granted_exclusive_not_gap();
-      if (!supremum_bit && rec_granted_exclusive_not_gap)
-        continue;
       if (UNIV_UNLIKELY(lock->type_mode & (LOCK_PREDICATE | LOCK_PRDT_PAGE)))
         continue; /* SPATIAL INDEX locking is broken. */
-      auto cell=
-        lock_sys.rec_hash.cell_get(lock->un_member.rec_lock.page_id.fold());
+      auto cell= lock_sys.rec_hash.cell_get(lock->un_member.rec_lock.
+                                            page_id.fold());
```

Comment:
I’d suggest that we assign the `fold` into a local variable, because we may be reusing it:
```c++
      const auto fold= lock->un_member.rec_lock.page_id.fold();
      auto cell= lock_sys.rec_hash.cell_get(fold);
```